src/table.ts

```diff
@@ -16,38 +16,37 @@
 
 'use strict';
 
-import {promisifyAll} from '@google-cloud/promisify';
+import { promisifyAll } from '@google-cloud/promisify';
 import * as is from 'is';
-import {ServiceError} from 'grpc';
+import { ServiceError } from 'grpc';
 import * as through from 'through2';
+import { Operation as GaxOperation } from 'google-gax/build/src/longrunning';
 
-import {Json} from './codec';
-import {Database} from './database';
-import {DatabaseAdminClient as d, SpannerClient as s} from './v1';
-import {PartialResultStream, Row} from './partial-result-stream';
-import {Snapshot, ReadRequest, Transaction, TimestampBounds} from './transaction';
+import { google as spanner_client } from '../proto/spanner';
+import { google as database_admin_client } from '../proto/spanner_database_admin';
 
-export type Key = string|string[];
+import { Json } from './codec';
+import { Database } from './database';
+import { PartialResultStream, Row } from './partial-result-stream';
+import { ReadRequest, TimestampBounds, Transaction, Snapshot } from './transaction';
+import { Schema } from './common';
 
-type Schema = string|string[]|{statements: string[], operationId?: string};
+export type Key = string | string[];
 
-type CommitPromise = Promise<[s.CommitResponse]>;
-type CreateTablePromise = Promise<[Table, d.Operation, d.GrpcOperation]>;
-type DropTablePromise = Promise<[d.Operation, d.GrpcOperation]>;
-type ReadPromise = Promise<[Array<Row|Json>]>;
-
-interface CreateTableCallback {
+export interface CreateTableCallback {
   (err: ServiceError, table?: null, operation?: null, apiResponse?: null): void;
-  (err: null, table: Table, operation: d.Operation,
-   apiResponse: d.GrpcOperation): void;
+  (err: null, table: Table, operation: GaxOperation,
+    apiResponse: DeleteResponse): void;
 }
-
-interface DropTableCallback {
+export type CreateTableResponse = [Table, GaxOperation, database_admin_client.longrunning.Operation];
```

Comment:
So just a general note, I'm seeing several places in the code were we reference the grpc message class object (like `longrunning.Operation`) but we actually don't return those anywhere. We should never use these types and instead use the interface counterparts `longrunning.IOperation`

Reply:
Yeah I was wondering about that - so would that instead be `GaxOperation` or would you rather it be `longrunning.IOperation`?

Reply:
Well I think the full response would be something like

```typescript
export type CreateTableResponse = [Table, GaxOperation, dbAdminClient.longrunning.IOperation];
```